### PR TITLE
Remove writeLockLastEdge

### PR DESCRIPTION
**Pull request by cheng-chang**

### CODE REVIEW

File: core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java [hunk 1]
```diff
@@ -174,9 +174,9 @@ public boolean isWrite() {
      * existing inodes: /a
      * last edge: a->b
      *
-     * @return whether the last edge should be write locked
+     * @return whether the last edge should be locked (either read or write lock)
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: if this includes read, why does the comparison not check for `READ`?

Comment (cheng-chang) [reply core/server/master/src/main/java/alluxio/master/file/meta/InodeTree.java]: I just removed the method, PTAL

**Comment by cheng-chang** [comment]
jenkins, test this please

File: core/server/master/src/main/java/alluxio/master/file/meta/LockedInodePath.java [hunk 2]
```diff
@@ -450,8 +450,8 @@ public void traverse() throws InvalidPathException {
           }
         }
         if (!nextInodeOpt.isPresent()) {
-          if (!mLockPattern.writeLockLastEdge()) {
-            // Other lock patterns only lock up to the last existing inode.
+          if (mLockPattern == LockPattern.READ) {
+            // Other lock patterns should lock the last non-existing edge.
```
Comment (gpang) [inline core/server/master/src/main/java/alluxio/master/file/meta/LockedInodePath.java]: "other" means WRITE_INODE and WRITE_EDGE? Maybe we should be a bit more explicit with this comment for clarity?

**Comment by cheng-chang** [comment]
alluxio-bot, merge this please

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Comment by calvinjia** [comment]
alluxio-bot, cherry-pick this to branch-2.0 please.

**Pull request merged**